linode/manager
# refactor: [M3-8186] - Clean up DebouncedSearchTextField and fix instances where debouncing is not happening

## Description 📝
This PR fixes the issue of debouncing is not happening for the instances of DebouncedSearchTextField component. 

## Changes  🔄
List any change relevant to the reviewer.
- Fix debounce issue in PlacementGroupsLanding.
- Fix debounce issue in LinodeSelectTable.
- Removed customeValue prop usage in LinodeSelectTable and SelectLinodePanel.
- Memoized the debounced onChange handler to prevent unnecessary re-creations. 
- Improved the `useEffect` hook to handle empty values and reduce unnecessary state updates. 
- removed redundant logic of the clear icon from the instanced of DebouncedSearchTextField. 

## Target release date 🗓️
09/02


## How to test 🧪

### Verification steps
(How to verify changes)
- Checkout the branch and verify there is no regression for all the references of `PlacementGroupsLanding`, `SelectLinodePanel` and `LinodeSelectTable`
- Verify debounce behavior for `PlacementGroupsLanding` and `LinodeSelectTable`
- Verify there is no regression in all the references of DebouncedSearchTextField.

## As an Author I have considered 🤔

*Check all that apply*
